### PR TITLE
Add common definition for SFP error handling

### PR DESCRIPTION
**Pull request by stephenxs**
Signed-off-by: Stephen Sun <stephens@nvidia.com>

### CODE REVIEW

File: sonic_platform_base/sfp_base.py [hunk 1]
```diff
@@ -16,6 +16,41 @@ class SfpBase(device_base.DeviceBase):
     # Device type definition. Note, this is a constant.
     DEVICE_TYPE = "sfp"
 
+    # Generic error types definition
+    SFP_STATUS_INITIALIZING = 'Initializing'
+    SFP_STATUS_OK = 'OK'
+    SFP_STATUS_UNPLUGGED = 'Unplugged'
+    SFP_STATUS_DISABLED = 'Disabled'
+    SFP_ERROR_DESCRIPTION_BLOCKING = 'Blocking Error'
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sfp_base.py]: Maybe a better name? "Blocking Error" sounds a little bit confuse.

File: sonic_platform_base/sfp_base.py [hunk 2]
```diff
@@ -16,6 +16,41 @@ class SfpBase(device_base.DeviceBase):
...
+    SFP_ERROR_DESCRIPTION_BAD_EEPROM = 'Bad or unsupported eeprom'
+    SFP_ERROR_DESCRIPTION_UNSUPPORTED_CABLE = 'Unsupported cable'
+    SFP_ERROR_DESCRIPTION_HIGH_TEMP = 'High temperature'
+    SFP_ERROR_DESCRIPTION_BAD_CABLE = 'Bad cable (module/cable is shorted)'
+
+    # SFP status
+    SFP_STATUS_BIT_REMOVED = 0
+    SFP_STATUS_BIT_INSERTED = 1
+    # SFP error status
+    SFP_ERROR_BIT_BLOCKING = 2
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sfp_base.py]: Maybe hex is more readable here, or could you please add comment at least?

File: sonic_platform_base/sfp_base.py [hunk 3]
```diff
@@ -384,4 +419,10 @@ def write_eeprom(self, offset, num_bytes, write_buffer):
         """
         raise NotImplementedError
 
+    def get_error_description(self):
+        """
+        Retrives the error status of the SFP module
 
+        Returns:
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sfp_base.py:426]: Could you please add description for the return value? I suppose if multiple error exists at the same time, "|" should be used to join them here.

File: sonic_platform_base/sfp_base.py (hunk 1 above)
Comment (stephenxs) [reply sonic_platform_base/sfp_base.py]: Fixed.

File: sonic_platform_base/sfp_base.py (hunk 2 above)
Comment (stephenxs) [reply sonic_platform_base/sfp_base.py]: Changed to hex.

File: sonic_platform_base/sfp_base.py (hunk 3 above)
Comment (stephenxs) [reply sonic_platform_base/sfp_base.py:426]: Added

**Review by Junchao-Mellanox — APPROVED** [review]

**Pull request closed without merging**